File: TrapprKeepr/Models/VaultKeep.cs
namespace TrapprKeepr.Models;

public class VaultKeep
{
    public int Id { get; set; }
    public int VaultId { get; set; }
    public int KeepId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string CreatorId { get; set; }
    public Profile Creator { get; set; }
}

File: TrapprKeepr.client/src/models/VaultKeep.js
import { Profile } from "./Account.js"

Profile

export class VaultKeep {
    constructor(data) {
        this.id = data.id
        this.vaultId = data.vaultId
        this.keepId = data.keepId
        this.creatorId = data.creatorId
        // NOTE maybe too much? not sure...but gonna try it and can take off if needed.
        this.profile = data.profile
    }
}